Insert on sqlite db with unique key cause a crash and exit of app (IOS)

### I have searched and made sure there are no existing issues for the issue I am filing

- [X] I have searched the existing issues

### Description

When i insert a duplicate record on sqlite db with unique key, app crash and exit.

### Expected Behavior

Response with and error and not exit and crash.

### Actual behavior

crash end exit with this error:
[ERROR] *** First throw call stack:
[ERROR] ( 
[ERROR]         0   CoreFoundation                      0x00007ff8004288ab __exceptionPreprocess + 242
[ERROR]         1   libobjc.A.dylib                     0x00007ff80004dba3 objc_exception_throw + 48
[ERROR]         2   CoreFoundation                      0x00007ff800428789 -[NSException initWithCoder:] + 0
[ERROR]         3   barberapp                           0x000000010a62569b -[TI_PLSqliteResultSet next] + 251
[ERROR]         4   barberapp                           0x000000010a6b7c0a -[TiDatabaseProxy executeSQL:withParams:withError:] + 282
[ERROR]         5   barberapp                           0x000000010a6b7e3b -[TiDatabaseProxy execute:] + 347
[ERROR]         6   CoreFoundation                      0x00007ff80042f2fc __invoking___ + 140
[ERROR]         7   CoreFoundation                      0x00007ff80042c529 -[NSInvocation invoke] + 305
[ERROR]         8   JavaScriptCore                      0x000000010f2826f0 _ZN3JSC24ObjCCallbackFunctionImpl4callEP9JSContextP13OpaqueJSValuemPKPKS3_PS6_ + 528
[ERROR]         9   JavaScriptCore                      0x000000010f28207c _ZN3JSCL34objCCallbackFunctionCallAsFunctionEPK15OpaqueJSContextP13OpaqueJSValueS4_mPKPKS3_PS6_ + 204
[ERROR]         10  JavaScriptCore                      0x000000010f28020e _ZN3JSCL24callObjCCallbackFunctionEPNS_14JSGlobalObjectEPNS_9CallFrameE + 574
[ERROR]         11  ???                                 0x00005b6851a04027 0x0 + 100503604183079
[ERROR]         12  JavaScriptCore                      0x000000010f205ecd llint_entry + 119186
[ERROR]         13  JavaScriptCore                      0x000000010f1e8b38 vmEntryToJavaScript + 216
[ERROR]         14  JavaScriptCore                      0x000000010fa2c258 _ZN3JSC11Interpreter11executeCallEPNS_14JSGlobalObjectEPNS_8JSObjectERKNS_8CallDataENS_7JSValueERKNS_7ArgListE + 552
[ERROR]         15  JavaScriptCore                      0x000000010f2a4bd1 JSObjectCallAsFunction + 545
[ERROR]         16  TitaniumKit                         0x000000010e364065 __65-[KrollObject invokeCallbackForKey:withObject:thisObject:onDone:]_block_invoke + 419
[ERROR]         17  libdispatch.dylib                   0x00007ff80013b7fb _dispatch_call_block_and_release + 12
[ERROR]         18  libdispatch.dylib                   0x00007ff80013ca3a _dispatch_client_callout + 8
[ERROR]         19  libdispatch.dylib                   0x00007ff80014c32c _dispatch_main_queue_drain + 1338
[ERROR]         20  libdispatch.dylib                   0x00007ff80014bde4 _dispatch_main_queue_callback_4CF + 31
[ERROR]         21  CoreFoundation                      0x00007ff800387b1f __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
[ERROR]         22  CoreFoundation                      0x00007ff800382436 __CFRunLoopRun + 2482
[ERROR]         23  CoreFoundation                      0x00007ff8003816a7 CFRunLoopRunSpecific + 560
[ERROR]         24  GraphicsServices                    0x00007ff809cb128a GSEventRunModal + 139
[ERROR]         25  UIKitCore                           0x000000011d379ad3 -[UIApplication _run] + 994
[ERROR]         26  UIKitCore                           0x000000011d37e9ef UIApplicationMain + 123
[ERROR]         27  barberapp                           0x000000010a61904b main + 1019
[ERROR]         28  dyld                                0x000000010ac942bf start_sim + 10
[ERROR]         29  ???                                 0x000000010ed2e52e 0x0 + 4543669550
[ERROR] ) 
[ERROR] undefined:116
[ERROR] (null)
[ERROR]                                     ^
[ERROR] : Error occurred calling next on a PLSqliteResultSet. SQLite error: 'UNIQUE constraint failed: codici.codice' for 'insert into codici (codice,nomedb,nome) values(?,?,?)'
[ERROR]     at  ([native code])
[ERROR]     at  ()
[ERROR]   
[ERROR]     CoreFoundation 0x00007ff8004288ab __exceptionPreprocess + 242
[ERROR]     libobjc.A.dylib 0x00007ff80004dba3 objc_exception_throw + 48
[ERROR]     CoreFoundation 0x00007ff800428789 -[NSException initWithCoder:] + 0
[ERROR]     barberapp 0x000000010a62569b -[TI_PLSqliteResultSet next] + 251
[ERROR]     barberapp 0x000000010a6b7c0a -[TiDatabaseProxy executeSQL:withParams:withError:] + 282
[ERROR]     barberapp 0x000000010a6b7e3b -[TiDatabaseProxy execute:] + 347
[ERROR]     CoreFoundation 0x00007ff80042f2fc __invoking___ + 140
[ERROR]     CoreFoundation 0x00007ff80042c529 -[NSInvocation invoke] + 305
[ERROR]     JavaScriptCore 0x000000010f2826f0 _ZN3JSC24ObjCCallbackFunctionImpl4callEP9JSContextP13OpaqueJSValuemPKPKS3_PS6_ + 528
[ERROR]     JavaScriptCore 0x000000010f28207c _ZN3JSCL34objCCallbackFunctionCallAsFunctionEPK15OpaqueJSContextP13OpaqueJSValueS4_mPKPKS3_PS6_ + 204
[ERROR]     JavaScriptCore 0x000000010f28020e _ZN3JSCL24callObjCCallbackFunctionEPNS_14JSGlobalObjectEPNS_9CallFrameE + 574
[ERROR]     ??? 0x00005b6851a04027 0x0 + 100503604183079
[ERROR]     JavaScriptCore 0x000000010f205ecd llint_entry + 119186
[ERROR]     JavaScriptCore 0x000000010f1e8b38 vmEntryToJavaScript + 216
[ERROR]     JavaScriptCore 0x000000010fa2c258 _ZN3JSC11Interpreter11executeCallEPNS_14JSGlobalObjectEPNS_8JSObjectERKNS_8CallDataENS_7JSValueERKNS_7ArgListE + 552
[ERROR]     JavaScriptCore 0x000000010f2a4bd1 JSObjectCallAsFunction + 545
[ERROR]     TitaniumKit 0x000000010e364065 __65-[KrollObject invokeCallbackForKey:withObject:thisObject:onDone:]_block_invoke + 419
[ERROR]     libdispatch.dylib 0x00007ff80013b7fb _dispatch_call_block_and_release + 12
[ERROR]     libdispatch.dylib 0x00007ff80013ca3a _dispatch_client_callout + 8
[ERROR]     libdispatch.dylib 0x00007ff80014c32c _dispatch_main_queue_drain + 1338
[ERROR] libc++abi: terminating with uncaught exception of type NSException

### Reproducible sample

        try {
        var querycod = db.execute('insert  into dbname (filed1,field2,field3) values(?,?,?)',data.value,data.value2,data.value3);
            
        } catch (error) {
        console.log(error);    
        }

### Steps to reproduce

just insert duplicate record

### Platform

iOS

### SDK version you are using

12.0.0

### Alloy version you are using

_No response_

--- a/iphone/Classes/TiDatabaseProxy.m
+++ b/iphone/Classes/TiDatabaseProxy.m
@@ -266,7 +266,15 @@
   }
 
   NSError *error = nil;
-  TiDatabaseResultSetProxy *result = [self executeSQL:sql withParams:params withError:&error];
+  TiDatabaseResultSetProxy *result = nil;
+
+  // Guard the native exception and re-throw, so it can be caught on the JavaScript side
+  @try {
+    result = [self executeSQL:sql withParams:params withError:&error];
+  } @catch (NSException *exception) {
+    [self throwException:@"failed to execute SQL statement" subreason:[error description] location:CODELOCATION];
+  }
+
   if (error != nil) {
     [self throwException:@"failed to execute SQL statement" subreason:[error description] location:CODELOCATION];
     return nil;

